platform/security/sts/security-sts-ldaplogin/src/main/java/ddf/ldap/ldaplogin/SslLdapLoginModule.java

```diff
@@ -335,6 +337,42 @@ protected boolean doLogin() throws LoginException {
         return true;
     }
 
+    void validateUsername(String username) throws LoginException {
```

Comment:
A little less efficient but easier to read would be:
```java
    private static final int[] BAD_CHARS = {'\\', ',', '+', '"', '<', '>', ';', '#'};

    void validateUsername(String username) throws LoginException {
        if (username.chars()
                .anyMatch(c -> Arrays.stream(BAD_CHARS)
                        .anyMatch(i -> i == c))) {
            throw new LoginException("Bad characters present in user name.");
        }
    }
```

Could also use a `Set<Integer>` and replace the nested stream/anyMatch with a `contains`.